### DISCUSSION TITLE
lettuce infinite print log - Unexpected exception during request: java.lang.NullPointerException

### DISCUSSION THREAD

**Discussion by wbzj1110** [redis/lettuce#1701] [Q&A]
# preface
When I encountered this problem, I found that the issue or discussion of lettuce did not have this problem. So I opened a discussion to record it for the convenience of those who encounter the same problem in the future.

I don't know if lettuce can fix this problem on the 5.3.x version to facilitate users who currently use spring-boot 2.3.9RELEASE. After all, the latest spring-boot realse that adapts to spring-cloud is as follows: 
https://github.com/spring-cloud/spring-cloud-release/wiki/Spring-Cloud-Hoxton-Release-Notes
![image](https://user-images.githubusercontent.com/11921135/113264198-47950480-9305-11eb-8b34-241efc6da85c.png)

![image](https://user-images.githubusercontent.com/11921135/113263938-fe44b500-9304-11eb-9d58-8d6821381e0a.png)

the issue as follows:
https://github.com/spring-projects/spring-boot/issues/25857

#  background

spring-boot 2.3.9 RELEASE
default lettuce 5.3.6 RELEASE
redis cluster mode  
![image](https://user-images.githubusercontent.com/11921135/113256647-0c420800-92fc-11eb-8591-57e49a48b165.png)

some bean use @PostConstruct  and method contain redis action ，This is the key to reproducing the problem。
``` 
@Service
@Slf4j
public class TestA   {


    @PostConstruct
    public void init(){
		//redis action
    }

}
```

# expression
log infinite print Unexpected exception during request: java.lang.NullPointerException，as follows：
![image](https://user-images.githubusercontent.com/11921135/113255584-94bfa900-92fa-11eb-8b3d-0fb2877cdf61.png)

 print very quickly。

# reasons
lettuce 5.3.6 RELEASE has a bug about use trace，the key code ，as follows：
![image](https://user-images.githubusercontent.com/11921135/113257311-03056b00-92fd-11eb-885b-e4d88a3e15df.png)

when use full link tracking related modules， lettuce does not first determine whether it is empty， and this is the reason。

# solutions
Either one will do

1）、pom add lettuce new version
lettuce fix it later，but in it new version，like 6.1.0.RELEASE，the code as follows：
https://github.com/lettuce-io/lettuce-core/blob/main/src/main/java/io/lettuce/core/protocol/CommandHandler.java#L460

![image](https://user-images.githubusercontent.com/11921135/113257926-dc93ff80-92fd-11eb-9518-bd37bf3bcef8.png)

2）、user code change
use @Order or  implement ApplicationRunner  rather than use  @PostConstruct 

# suggest

1)  spring-boot 
spring-boot 2.3.9 release may use new version of lettuce，at least not lettuce 5.3.6 RELEASE 。
https://github.com/spring-projects/spring-boot/issues/25857#issuecomment-811720102
Obviously spring-boot needs to wait for lettuce to solve this problem. 

2）lettuce fix 5.3.6 RELEASE this bug。

**Comment by mp911de** [comment]
There's indeed a potential loophole that commands may not be wrapped into `TracedCommand`. Do you want to submit a pull request to fix the issue on `main`? We'll take care of the backports.

**Reply by wbzj1110** [reply]
OK，I will do it later。☺

**Reply by wbzj1110** [reply]
https://github.com/lettuce-io/lettuce-core/pull/1703 
 hi @mp911de  this pull request has been merged.   Are there any plans to build the new 5.3.X RELEASE ?   
Spring-boot newest RELEASE  is 2.3.9  .it has this bug.  As follows spring-boot need us to build new version,and they will update the dependency version.  
https://github.com/spring-projects/spring-boot/issues/25857

**Reply by mp911de** [reply]
We can ship a release by next week.

**Reply by wbzj1110** [reply]
OK，thks~
